app/controllers/signup.js

```diff
@@ -1,291 +1,55 @@
-      type: 'text',
-      errorMessage: '',
-      required: false,
-      showError: false,
-    },
-    {
-      id: 'twitter_id',
-      label: 'Twitter username (Not the full URL)',
-      placeholder: 'anakin7',
-      type: 'text',
-      errorMessage: 'Twitter handle is required',
-      required: true,
-      showError: false,
-    },
-    {
-      id: 'website',
-      label: 'Website (Optional)',
-      placeholder: 'Your portfolio website if any. e.g mysonisajedi.com',
-      type: 'text',
-      errorMessage: '',
-      required: false,
-      showError: false,
-    },
-  ];
-
-  timerId = undefined;
-
-  //Debounce Function
-  debounce(func, delay) {
-    clearTimeout(this.timerId);
-    this.timerId = setTimeout(func, delay);
-  }
-
-  @action handleFieldChange(name, value) {
-    const index = this.fields.findIndex((field) => field.id === name);
-    set(this.formData, name, value);
-
-    if (this.fields[index].required && !value) {
-      this.isSubmitDisabled = true;
-      set(this.fields[index], 'showError', true);
-    } else if (!this.fields[index].validator) {
-      set(this.fields[index], 'showError', false);
-    }
-
-    const anyErrors = this.fields.map((field) => {
-      if (field.required && this.formData[field.id] === '') {
-        return true;
-      }
-      return false;
-    });
-
-    if (anyErrors.filter(Boolean).length) {
-      this.isSubmitDisabled = true;
-    } else {
-      this.isSubmitDisabled = false;
-    }
-  }
-
-  async checkUserName(userName) {
-    if (!userName) {
-      return set(this.fields[2], 'errorMessage', 'Username cannot be empty');
-    }
-    try {
-      const lowerCaseUsername = userName.toLowerCase();
-      const response = await fetch(
-        `${BASE_URL}/users/isUsernameAvailable/${lowerCaseUsername}`,
-        {
-          method: 'GET',
-          headers: {
-            'Content-Type': 'application/json',
-          },
-          credentials: 'include',
-        }
-      );
-      const data = await response.json();
-      const { isUsernameAvailable } = data;
-      set(this.fields[2], 'showError', !isUsernameAvailable);
-      set(
-        this.fields[2],
-        'errorMessage',
-        `${userName} is not available, Choose another username!`
-      );
-    } catch (error) {
-      console.error('Error : ', error);
-    }
-  }
-
-  @action async userNameValidator(userName) {
-    this.debounce(() => this.checkUserName(userName), 500);
-  }
-
-  @action phoneNumberValidator(phone) {
-    if (typeof phone !== 'string') {
-      return false;
-    }
-
-    const pattern = /^(0|[+91]{3})?[7-9][0-9]{9}$/;
-    const index = this.fields.findIndex((field) => field.id === 'phone');
-
-    if (pattern.test(phone)) {
-      set(this.fields[index], 'showError', false);
-    } else {
-      set(this.fields[index], 'showError', true);
-    }
+  @action changeRouteParams(paramValue) {
+    if (paramValue)
+      this.transitionToRoute({ queryParams: { state: paramValue } });
```

Comment:
Use `router` service because `this.transitionToRoute` is deprecated.
https://deprecations.emberjs.com/v3.x/#toc_routing-transition-methods